pkg/network/usm/usm_http2_monitor_test.go

```diff
@@ -521,3 +584,55 @@ func createMessageWithPingAndWindowUpdate(t *testing.T, headerFields []hpack.Hea
 
 	return buf.Bytes()
 }
+
+// createMessageWithRST creates a message with the given number of header frames and RST frames,
+// and inserts RST frames with the given error code.
+func createMessageWithRST(t *testing.T, headerFields []hpack.HeaderField, headerFramesCount, rstFramesCount, rstFramesCountInMessage int, errCode http2.ErrCode, includeStatusCode ...bool) []byte {
+	var buf bytes.Buffer
+	framer := http2.NewFramer(&buf, nil)
+
+	headersFrame, err := usmhttp2.NewHeadersFrameMessage(headerFields)
+	require.NoError(t, err, "could not create headers frame")
+
+	for i := 0; i < headerFramesCount; i++ {
+		streamID := 2*i + 1
+
+		// Writing the header frames to the buffer using the Framer.
+		require.NoError(t, framer.WriteHeaders(http2.HeadersFrameParam{
+			StreamID:      uint32(streamID),
+			BlockFragment: headersFrame,
+			EndStream:     false,
+			EndHeaders:    true,
+		}), "could not write header frames")
+
+		if len(includeStatusCode) > 0 && includeStatusCode[0] {
+			statusOkHeaderField := []hpack.HeaderField{
+				{Name: ":status", Value: "200"},
+			}
```

Comment:
that's a server response, how can you write it as a client?

Reply:
removed